src/nbody_simulation_result.jl

```diff
@@ -264,27 +273,38 @@ function total_energy(sr::SimulationResult, time::Real)
 end
 
 function initial_energy(simulation::NBodySimulation)
-    (u0, v0, n) = gather_bodies_initial_coordinates(simulation.system)
+    (u0, v0, n) = gather_bodies_initial_coordinates(simulation)
     ms = get_masses(simulation.system)
     return potential_energy(u0, simulation) + kinetic_energy(v0, ms)
 end
 
-# Instead of treating NBodySimulation as a DiffEq problem and passing it into a solve method
-# it is better to use a specific function for n-body simulations.
-function run_simulation(s::NBodySimulation, alg_type=Tsit5(), args...; kwargs...)
+function run_simulation(s::NBodySimulation, args...; kwargs...)
+    if s.thermostat isa LangevinThermostat
+        calculate_simulation_sde(s, args...; kwargs...)
+    else
+        calculate_simulation(s, args...; kwargs...)
+    end
+end
+
+function calculate_simulation(s::NBodySimulation, alg_type=Tsit5(), args...; kwargs...)
     solution = solve(ODEProblem(s), alg_type, args...; kwargs...)
     return SimulationResult(solution, s)
 end
 
 # this should be a method for integrators designed for the SecondOrderODEProblem (It is worth somehow to sort them from other algorithms)
-function run_simulation(s::NBodySimulation, alg_type::Union{VelocityVerlet,DPRKN6,Yoshida6}, args...; kwargs...)
+function calculate_simulation(s::NBodySimulation, alg_type::Union{VelocityVerlet,DPRKN6,Yoshida6}, args...; kwargs...)
```

Comment:
File an issue with OrdinaryDiffEq.jl asking for better algorithm categories for the Symplectic / RKN solvers